lib/puppet/pops/types/puppet_object.rb

```diff
@@ -11,6 +11,16 @@ module PuppetObject
   def _ptype
     self.class._ptype
   end
+
+  def _pall_contents(path, &block)
+  end
+
+  def _pcontents
+  end
+
+  def i12n_hash
```

Comment:
This is a nasty i14e name (incomprehensible). It is also part of every AST's API.